src/gc-arena/src/context.rs

```diff
@@ -165,6 +163,10 @@ impl Context {
     // In order for this to be safe, at the time of call no `Gc` pointers can be live that are not
     // reachable from the given root object.
     pub(crate) unsafe fn do_collection<R: Collect>(&self, root: &R, work: f64) -> f64 {
+        self.do_collection_inner(root, work)
+    }
+
+    fn do_collection_inner<R: Collect>(&self, root: &R, work: f64) -> f64 {
```

Comment:
Does this need to be a separate function?

Reply:
This is here so that the collection code isn't inside a big `unsafe` block.